src/main/java/io/sqreen/powerwaf/ByteBufferSerializer.java

```diff
@@ -397,26 +408,41 @@ public void close() {
 
     static class PWArgsSegment {
         ByteBuffer buffer;
+        List<PWArgsArrayBuffer> pwargsArrays = new ArrayList<>();
```

Comment:
If this is just an object pool for recycling `PWArgsArrayBuffer` objects, perhaps it would make more sense to have at a more global level than just per segment

Reply:
Yes, that can be an additional optimization. I left that out for now.